src/lib/core/tests/TestCHIPTLV.cpp

```diff
@@ -4051,6 +4077,114 @@ static void CheckGetByteView(nlTestSuite * inSuite, void * inContext)
     }
 }
 
+static void CheckChainedBuffers(nlTestSuite * inSuite, void * inContext)
+{
+    chip::System::TLVPacketBufferBackingStore store;
+    chip::System::PacketBufferHandle buffer;
+    CHIP_ERROR err;
+    uint8_t * data;
+    uint32_t bufSize = System::PacketBuffer::kMaxSizeWithoutReserve * 2;
+
+    buffer = chip::System::PacketBufferHandle::New(System::PacketBuffer::kMaxSizeWithoutReserve, 0);
+    NL_TEST_ASSERT(inSuite, !buffer.IsNull());
+
+    //
+    // Enable chaining during initialization of the store
+    //
+    store.Init(std::move(buffer), true);
+
+    //
+    // Allocate a scratch buffer that exceeds the size of a single packet buffer so that we
+    // test out the chaining capability.
+    //
+    data = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(bufSize));
+    NL_TEST_ASSERT(inSuite, data);
+
+    //
+    // Write out large byte strings and char strings that overflow a single packet buffer
+    // and ensure it succeeds.
+    //
+    {
+        TLV::TLVWriter writer;
+
+        writer.Init(store);
+
+        err = writer.PutBytes(TLV::CommonTag(0), data, bufSize);
+        NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+
+        err = writer.PutString(TLV::CommonTag(1), chip::Uint8::to_const_char(data), bufSize);
+        NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+
+        err = writer.Finalize();
+        NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    }
+
+    //
+    // Confirm that we can copy out the large byte string
+    // from the underlying chained backing buffers.
+    //
+    {
+        TLV::TLVReader reader;
+        chip::System::PacketBufferHandle tmpBuf;
+
+        tmpBuf = store.Release();
+        store.Init(std::move(tmpBuf), true);
+
+        reader.Init(store);
+
+        err = reader.Next();
+        NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+
+        err = reader.GetBytes(data, bufSize);
```

Comment:
Need to verify that the data and the size are as expected?

Reply:
The exciting thing about `GetBytes` is that you have no way to know how many bytes it actually grabbed, apart from calling `GetLength()` before calling `GetBytes`.